MapboxCoreNavigationTests/OfflineRoutingTests.swift

```diff
@@ -0,0 +1,125 @@
+
+        XCTAssertNotNil(route)
+        XCTAssertEqual(route!.coordinates!.count, 47)
+    }
+    
+    func testOfflineDirectionsError() {
+        
+        let bundle = Bundle(for: Fixture.self)
+        let tilesURL = URL(fileURLWithPath: bundle.bundlePath).appendingPathComponent("/tiles/liechtenstein")
+        
+        let setupExpectation = expectation(description: "Set up offline routing")
+        
+        let directions = NavigationDirections(accessToken: "foo")
+        directions.configureRouter(tilesURL: tilesURL) { (numberOfTiles) in
+            XCTAssertEqual(numberOfTiles, 5)
+            setupExpectation.fulfill()
+        }
+        
+        wait(for: [setupExpectation], timeout: 2)
+        
+        // Coordinates in SF
+        let coordinates = [CLLocationCoordinate2D(latitude: 37.7870, longitude: -122.4261),
+                           CLLocationCoordinate2D(latitude: 37.7805, longitude: -122.4073)]
+        
+        let options = NavigationRouteOptions(coordinates: coordinates, profileIdentifier: .automobile)
+        let calculateRouteExpectation = expectation(description: "Calculate route offline")
+        
+        directions.calculate(options, offline: true) { (waypoints, routes, error) in
+            XCTAssertNotNil(error)
+            XCTAssertEqual(error!.localizedDescription, "No suitable edges near location")
+            XCTAssertNil(routes)
+            XCTAssertNil(waypoints)
+            calculateRouteExpectation.fulfill()
+        }
+        
+        wait(for: [calculateRouteExpectation], timeout: 2)
+    }
+    
+    func testUnpackTilePack() {
+        
+        let bundle = Bundle(for: Fixture.self)
+        let readonlyPackURL = URL(fileURLWithPath: bundle.path(forResource: "li", ofType: "tar")!)
+        
+        let documentDirectory = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true)[0]
+        
+        // Copy the read-only tar file so navigator can consume it
+        let data = try! Data(contentsOf: readonlyPackURL)
+        let packURL = URL(fileURLWithPath: documentDirectory, isDirectory: true).appendingPathComponent(readonlyPackURL.lastPathComponent)
+        try! data.write(to: packURL)
+        
+        let outputDirectoryURL = URL(fileURLWithPath: documentDirectory, isDirectory: true).appendingPathComponent("tiles/test")
+        try? FileManager.default.createDirectory(atPath: outputDirectoryURL.path, withIntermediateDirectories: true, attributes: nil)
+        
+        var progressUpdates = 0
+        let unpackExpectation = self.expectation(description: "Tar file should be unpacked")
+        let progressExpectation = self.expectation(description: "Progress should be reported")
+        
+        NavigationDirections.unpackTilePack(at: packURL, outputDirectoryURL: outputDirectoryURL, progressHandler: { (totalBytes, unpackedBytes) in
+            
+            progressUpdates += 1
+            
+            if (progressUpdates >= 2) {
```

Comment:
XCTTestExpectation has an `expectedFulfillmentCount` property that we can use instead of defining our own counter.